src/item-navigator.ts

```diff
@@ -154,13 +168,27 @@ export class ItemNavigator
     `;
   }
 
+  slotChange(e: Event, type: 'header' | 'main'): void {
+    const slottedContent = (e.target as HTMLSlotElement).assignedNodes()?.[0] as HTMLElement;
+
+    this.dispatchEvent(
+      new CustomEvent('slotChange', {
+        detail: { slot: slottedContent, type },
+      })
+    );
+    this.requestUpdate();
+  }
+
   render(): TemplateResult {
     const displayReaderClass = this.loaded ? '' : 'hidden';
     return html`
       <div id="frame" class=${`${this.menuClass}`}>
+        <slot
```

Comment:
same here, not sure if the positioning of `>` end tags were from the linting/pre-commit formatting... but is it possible to have it positioned in a way that it's still somehow coupled with it's element?

Reply:
ah, I see end tag wasn't updated with pre-commit hook.. hmm, good catch. will indent closing tag to align.

Reply:
hmm, looks like they are, actually.. are you asking about putting this element all in one line?

Reply:
i was wondering if it can be in the same line something like:
```
<slot
  name="theater-header"
  @slotchange=${(e: Event) => this.slotChange(e, 'header')}>
</slot>
```